test/dedup-id/src/test/test.js

```diff
@@ -0,0 +1,207 @@
+const util = require('util');
+const uuid = require('uuid');
+
+const debug = require('abacus-debug')('abacus-dedup-id-test');
+
+const doGet = util.promisify(request.get);
+const doPost = util.promisify(request.post);
+
+const systemClientId = process.env.CLIENT_ID;
+const systemClientSecret = process.env.CLIENT_SECRET;
+const authServerURL = process.env.AUTH_SERVER_URL || 'http://localhost:9882';
+const collectorURL = process.env.COLLECTOR_URL || 'http://localhost:9080';
+const reportingURL = process.env.REPORTING_URL || 'http://localhost:9088';
+const pollInterval = process.env.POLL_INTERVAL || 300;
+
+const localMeterURL = 'http://localhost:9100';
+
+describe('dedup acceptance test', () => {
+  const timestamp = moment.now();
+  const testQuantity = 1;
+  const dedupId = 'test-dedup-id';
+  const differentDedupId = 'different-dedup-id';
+
+  let docWithoutDedupId;
+  let docWithDedupId;
+  let orgId;
+
+  let systemToken;
+
+  const secured = () => process.env.SECURED === 'true';
+
+  const authHeader = (token) => {
+    return token ?
+      {
+        headers: {
+          authorization: token()
+        }
+      } :
+      {};
+  };
+
+  const buildUsageDoc = (orgID, dedupId) => {
+    const usageDoc = {
+      start: timestamp,
+      end: timestamp,
+      organization_id: orgID,
+      space_id: 'test-dedup-id-space-id',
+      consumer_id: 'test-dedup-id-consumer-id',
+      resource_id: 'object-storage',
+      plan_id: 'basic',
+      resource_instance_id: 'test-dedup-id-resource-instance-id',
+      measured_usage: [{
+        measure: 'heavy_api_calls',
+        quantity: testQuantity
+      }]
+    };
+
+    if (dedupId)
+      extend(usageDoc, {
+        dedup_id: dedupId
+      });
+
+    return usageDoc;
+  };
+
+  // On local environment GET by location header is not routed to meter
+  const buildCorrectLocationHeaderUrl = (url) => {
+    if(url.indexOf('localhost') > -1)
+      return localMeterURL + url.substring(url.indexOf('v1/') - 1, url.length);
+    return url;
+  };
+
+  const sleep = (duration) => {
+    return new Promise((cb) => setTimeout(cb, duration));
+  };
+
+  const eventually = async (func) => {
+    while (true) {
+      try {
+        return await func();
+      } catch (e) {
+        debug('not ready yet: %o', e.message);
+      }
+      await sleep(pollInterval);
+    }
+  };
+
+  const sendUsage = async (usage) => {
+    const resp = await doPost(collectorURL + '/v1/metering/collected/usage',
+      extend({ body: usage }, authHeader(systemToken)));
+
+    expect(resp.statusCode).to.equal(202);
+    return buildCorrectLocationHeaderUrl(resp.headers.location);
+  };
+
+  before((done) => {
+    if(!secured())
+      done();
+    else {
+      systemToken = oauth.cache(authServerURL, systemClientId, systemClientSecret,
+        'abacus.usage.read abacus.usage.write'
+      );
+
+      systemToken.start((err) => {
+        if (err) debug('Unable to obtain system oAuth token due to %o', err);
```

Comment:
You should throw an Error here! No need to have the test continue execution if Abacus is secured and you have not managed to obtain a token.
You could have also considered a way to use async by converting `start` to Promise, if possible.